2-Draft-Accepted/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -20,29 +20,65 @@ There is no easy way in PowerShell to visualize a markdown document on console.
-Optionally, the output can be formatted as HTML.
+This RFC proposes to use `VT100` escape sequences to render markdown content.
+`ConvertFrom-Markdown` cmdlet as part of `Microsoft.PowerShell.Utility` PowerShell module would consume a string or a file path and output a PSObject containing three properties, `Tokens`, `Html`, `VT100EncodedString`.
+The `Tokens` property has the AST for the markdown document from `markdig`.
+The `Html` property has the markdown document converted to `HTML`.
+The `VT100EncodedString` property has the markdown documented with `VT100` escape sequences.
+By default, the `Tokens`and `Html` properties will be populated.
+The `VT100EncodedString` property will be populated only when `-AsVT100EncodedString` is specified.
+
 For converting strings to VT100 coded strings, we will be writing an extension to [markdig](https://github.com/lunet-io/markdig).
 The extension will insert VT100 escape sequences as appropriate.
 
+### Specification for `ConvertFrom-Markdown`
+
 ```PowerShell
-ConvertFrom-Markdown [-Path] <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
 
-ConvertFrom-Markdown -LiteralPath <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+ConvertFrom-Markdown [-Path] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-LiteralPath] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-InputObject] <psobject> [-AsVT100EncodedString] [<CommonParameters>]
 
-ConvertFrom-Markdown -InputObject <psobject> -AsHTML -AsPlainText  [<CommonParameters>]
 ```
 
-### Parameters
+#### Parameters
 
 - **Path** : Accepts an array of file paths with markdown content.
 - **LiteralPath** : Accepts an array of literal paths with markdown content.
-- **InputObject** : Accepts an InputObject of `System.IO.FileSystemInfo`, `string` type.
-- **AsHTML** : When selected, the output will be HTML.
-- **AsPlainText** : When selected, the output will be plain text.
+- **InputObject** : Accepts an InputObject of `System.IO.FileInfo`, `string` type.
+- **AsVT100EncodedString** : When selected, the `VT100EncodedString` property is populated.
+
+There will be support for changing the colors for various elements using the `Set-MarkdownOption` cmdlet. To retrieve the current settings `Get-MarkdownOption` cmdlet can be used.
+
+### Specification for `Set-MarkdownOption`
+
+```PowerShell
+
+Set-MarkdownOption [-Header1Color <ConsoleColor>] [-Header2Color <ConsoleColor>] [-Header3Color <ConsoleColor>] [-Header4Color <ConsoleColor>] [-Header5Color <ConsoleColor>] [-Header6Color <ConsoleColor>] [-CodeBlockForegroundColor <ConsoleColor>] [-CodeBlockBackgroudColor <ConsoleColor>] [-ImageAltTextForegroundColor <ConsoleColor>] [-LinkForegroundColor <ConsoleColor>] [-ItalicsForegroundColor <ConsoleColor>] [<CommonParameters>]
+
+Set-MarkdownOption [-Theme <MarkdownThemeType>] [<CommonParameters>]
```

Comment:
I think about custom themes -  using ThemeName as string is more general.
If we store the theme type/name in `powershell.config.json` I believe the cmdlet is unnecessary - I'd prefer to have one universal cndlet to edit `powershell.config.json` (out of the RFC).
I believe `Set-MarkdownOption [-Header1Color <ConsoleColor>] ... ` is unnecessaty too - how often should users change a theme? I suppose very seldom. If we store a theme in a simple file (json/yaml?) - why we don't address the file format/place in the RFC? - users can manually edit it easily.

Reply:
> If we store the theme type/name in powershell.config.json

Having the theme setting in `powershell.config.json` doesn't seem necessary. I think having `Set-MarkdownOption -Theme <name>` in your profile would work.

> If we store a theme in a simple file (json/yaml?) - why we don't address the file format/place in the RFC?

Do you mean to allow a user to alter a theme and create their own theme?

Reply:
I modelled it after `Set-PSReadLineOption`. And as @daxian-dbw said, all the customization can be added to the profile.

Reply:
>Do you mean to allow a user to alter a theme and create their own theme?

Yes.

>all the customization can be added to the profile.

I understand. I still think that customizing themes is a rare scenario and creating special cmdlets is redundant. It is also easier to create and distribute custom themes in a file than in a profile.

Reply:
@iSazonov I have updated the RFC to supported importing / exporting the settings to/from a file.